src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,115 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static int CAPACITY = 10;
+    private static double SIZE_OF_INCREASE = 1.5;
+    private T[] values;
+    private int size;
+
+    public ArrayList() {
+        values = (T[]) new Object[CAPACITY];
+    }
+
+    private void increaseCapacity() {
+        if (values.length == size) {
+            int newCapacity = (int)(values.length * SIZE_OF_INCREASE);
+            values = Arrays.copyOf(values, newCapacity);
+        }
+    }
+
+    private void correctIndex(int index) {
+        if (index < 0 || index >= size) {
+            throw new ArrayListIndexOutOfBoundsException("Index: " + index + ", Size: " + size);
+        }
```

Comment:
The name 'correctIndex' does not clearly describe the method's purpose. A name like 'validateIndex' or 'checkIndexBounds' would be more informative.